### ISSUE TITLE
[BUG] Deep level aggregations query hang the request

### ISSUE THREAD

**Issue by Roboteus** [opensearch-project/OpenSearch#15914]
### Describe the bug

Application in **version 2.16.0 has a bug** which is manifested by hanging while trying to resolve the request - wait for unlimited amount of time. The problem is only for specific query which is included to this bug ticket below.
**Version 1.3.19 handling the query**

### Related component

Search

### To Reproduce

1. Create index:
```
PUT supplier2/
{
  "settings": {
    "number_of_shards": 1,
    "analysis": {
      "analyzer": {
        "default": {
          "tokenizer": "whitespace",
          "filter": [
            "lowercase"
          ]
        }
      },
      "normalizer": {
        "raw_normalizer": {
          "type": "custom",
          "filter": [
            "lowercase"
          ]
        }
      }
    }
  },
  "mappings": {
    "properties": {
      "dataSet": {
        "type": "keyword"
      },
      "supplierValueProperties": {
        "type": "nested",
        "properties": {
          "propertyName": {
            "type": "keyword"
          },
          "propertyType": {
            "type": "keyword"
          },
          "propertyStringValue": {
            "type": "text",
            "fields": {
              "raw": {
                "type": "keyword"
              },
              "raw_normalized": {
                "type": "keyword",
                "normalizer": "raw_normalizer"
              }
            }
          },
          "propertyNumericValue": {
            "type": "double"
          },
          "propertyDateValue": {
            "type": "date",
            "format": "yyyy-MM-dd"
          },
          "propertyBooleanValue": {
            "type": "boolean"
          }
        }
      },
      "references": {
        "type": "nested",
        "properties": {
          "key": {
            "type": "keyword"
          },
          "value": {
            "type": "nested",
            "properties": {
              "id": {
                "type": "keyword"
              },
              "code": {
                "type": "text",
                "fields": {
                  "raw": {
                    "type": "keyword"
                  },
                  "raw_normalized": {
                    "type": "keyword",
                    "normalizer": "raw_normalizer"
                  }
                }
              },
              "referenceValueProperties": {
                "type": "nested",
                "properties": {
                  "propertyName": {
                    "type": "keyword"
                  },
                  "propertyType": {
                    "type": "keyword"
                  },
                  "propertyStringValue": {
                    "type": "text",
                    "fields": {
                      "raw": {
                        "type": "keyword",
                        "ignore_above": 30000
                      },
                      "raw_normalized": {
                        "type": "keyword",
                        "normalizer": "raw_normalizer",
                        "ignore_above": 30000
                      }
                    }
                  },
                  "propertyNumericValue": {
                    "type": "double"
                  },
                  "propertyDateValue": {
                    "type": "date",
                    "format": "yyyy-MM-dd"
                  },
                  "propertyBooleanValue": {
                    "type": "boolean"
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```
2. Execute the query:
```
POST supplier2/_search
{
  "size": 1000,
  "query": {
    "bool": {
      "must": [
        {
          "term": {
            "dataSet": {
              "value": "basic",
              "boost": 1
            }
          }
        }
      ],
      "adjust_pure_negative": true,
      "boost": 1
    }
  },
  "aggregations": {
    "reference_aggregation": {
      "nested": {
        "path": "references"
      },
      "aggregations": {
        "references.key": {
          "terms": {
            "field": "references.key"
          },
          "aggregations": {
            "referenceValueProperties": {
              "nested": {
                "path": "references.value.referenceValueProperties"
              },
              "aggregations": {
                "propertyName": {
                  "terms": {
                    "field": "references.value.referenceValueProperties.propertyName"
                  },
                  "aggregations": {
                    "propertyType": {
                      "terms": {
                        "field": "references.value.referenceValueProperties.propertyType"
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

### Expected behavior

request resolved, like in this version:
  "version" : {
    "distribution" : "opensearch",
    "number" : "1.3.19",
    "build_type" : "zip",
    "build_date" : "2024-08-23T00:39:31.484729800Z",
    "build_snapshot" : false,
    "lucene_version" : "8.10.1",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  }

### Additional Details

default installation, for example in Windows (not related to OS):
"version" : {
    "distribution" : "opensearch",
    "number" : "2.16.0",
    "build_type" : "zip",
    "build_date" : "2024-08-06T20:32:32.086481300Z",
    "build_snapshot" : false,
    "lucene_version" : "9.11.1",
    "minimum_wire_compatibility_version" : "7.10.0",
    "minimum_index_compatibility_version" : "7.0.0"
}

**Comment by kkewwei** [comment]
@Roboteus, it to be related to #13324, I will find the reason and fix it as soon as possible.

**Comment by msfroh** [comment]
> The issue seems to be not fixed [1], the branch in question included the supposed fix:

@reta -- I notice that it's failing on a MixedClusterClientYamlTestSuiteIT. Do you think it might be a result of the old node that doesn't have the fix yet?

I noticed the `skip` setting is:
```
"Supported queries":
  - skip:
      version: " - 2.17.99"
      reason: "fixed in 2.18.0"
```

Maybe that should be `- 2.99.99` until we backport the fix to 2.x.

**Comment by reta** [comment]
> > The issue seems to be not fixed [1], the branch in question included the supposed fix:
> 
> @reta -- I notice that it's failing on a MixedClusterClientYamlTestSuiteIT. Do you think it might be a result of the old 2.18 node that doesn't have the fix yet?
> 
> I noticed the `skip` setting is:
> ```
> "Supported queries":
>   - skip:
>       version: " - 2.17.99"
>       reason: "fixed in 2.18.0"
> ```
> 
> Maybe that should be `- 2.99.99` until we backport the fix to 2.x.

Could be it since no backports happened, thanks @msfroh

**Comment by kkewwei** [comment]
> backports
@reta, If we should change the skip like this:

```
"Supported queries":
  - skip:
      version: " - 2.99.99"
      reason: "fixed in 3.0.0"
```

This case seems happen a bit high frequently.

**Comment by reta** [comment]
@kkewwei if this is bwc issue, the backport to 2.x should fix it, could you please backport manually (if it makes sense) since auto backport failed https://github.com/opensearch-project/OpenSearch/pull/15931#issuecomment-2360398545. Thank you
